### PR TITLE
Message recovery function returns Vec of results instead of failing when one or more measurements are corrupt

### PR DESCRIPTION
**Pull request by DJAndries**
When calling `recover` with one or more corrupt messages, the function will return a single error for all measurements. This makes it difficult to determine which message is corrupt. This PR changes the return type to a Vec of results to mitigate this issue.

### CODE REVIEW

**Comment by rillian** [comment]
Can you say more about the issue you're trying to address here?

**Comment by DJAndries** [comment]
@rillian edited description

File: src/internal.rs [hunk 1]
```diff
@@ -527,24 +533,21 @@
...
-  let measurement = &splits[0].as_ref().unwrap().0;
-  for to_chk in splits.iter().skip(1) {
-    let measurement_to_chk = &to_chk.as_ref().unwrap().0;
-    if measurement_to_chk != measurement {
-      return Err(Error::ClientMeasurementMismatch(
-        serde_json::to_string(measurement).unwrap(),
-        serde_json::to_string(&measurement_to_chk).unwrap(),
-      ));
+  if let Some(measurement) = &splits.iter().find(|res| res.is_ok()) {
+    let measurement = &measurement.as_ref().unwrap().0;
```
[the remark this answers is not in the document]
Comment (rillian) [reply src/internal.rs:549]: `unwrap` is ok here; we shouldn't have a `None` measurement not in the splits.

File: src/internal.rs [hunk 2]
```diff
@@ -527,23 +543,18 @@ pub fn recover(
...
+    for to_chk in splits.iter().flatten() {
+      let measurement_to_chk = &to_chk.0;
+      if measurement_to_chk != measurement {
+        return Err(Error::ClientMeasurementMismatch(
+          serde_json::to_string(measurement).unwrap(),
+          serde_json::to_string(&measurement_to_chk).unwrap(),
+        ));
```
Comment (rillian) [inline src/internal.rs:556]: Would it make sense to stuff the error back into the `splits` vec, so this could return one less layer of `Result`?

**Review by rillian — APPROVED** [review]
This is...not beautiful code. Seems like a reasonable approach in context.

**Comment by DJAndries** [comment]
> This is...not beautiful code. Seems like a reasonable approach in context.

tbh, the whole repo could use some tlc

**Review by rillian — APPROVED** [review]
Yay! This is a nice cleanup.

**Pull request merged**